// @capacitor-firebase/analytics/dist/plugin.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@capacitor/core');
var analytics = require('firebase/analytics');

const FirebaseAnalytics = core.registerPlugin('FirebaseAnalytics', {
    web: () => Promise.resolve().then(function () { return web; }).then(m => new m.FirebaseAnalyticsWeb()),
});

class FirebaseAnalyticsWeb extends core.WebPlugin {
    async getAppInstanceId() {
        throw this.unimplemented('Not implemented on web.');
    }
    async setUserId(options) {
        const analytics$1 = analytics.getAnalytics();
        analytics.setUserId(analytics$1, options.userId || '');
    }
    async setUserProperty(options) {
        const analytics$1 = analytics.getAnalytics();
        analytics.setUserProperties(analytics$1, {
            [options.key]: options.value,
        });
    }
    async setCurrentScreen(options) {
        const analytics$1 = analytics.getAnalytics();
        analytics.logEvent(analytics$1, 'screen_view', {
            firebase_screen: options.screenName || undefined,
            firebase_screen_class: options.screenClassOverride || undefined,
        });
    }
    async logEvent(options) {
        const analytics$1 = analytics.getAnalytics();
        analytics.logEvent(analytics$1, options.name, options.params);
    }
    async setSessionTimeoutDuration(_options) {
        throw this.unimplemented('Not implemented on web.');
    }
    async setEnabled(_options) {
        const analytics$1 = analytics.getAnalytics();
        analytics.setAnalyticsCollectionEnabled(analytics$1, _options.enabled);
    }
    async isEnabled() {
        const enabled = window['ga-disable-analyticsId'] === true;
        return {
            enabled,
        };
    }
    async resetAnalyticsData() {
        throw this.unimplemented('Not implemented on web.');
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    FirebaseAnalyticsWeb: FirebaseAnalyticsWeb
});

exports.FirebaseAnalytics = FirebaseAnalytics;
